src/js/packages/@reactpy/client/src/vdom.tsx

```diff
@@ -103,6 +105,27 @@ function createImportSourceElement(props: {
   );
 }
 
+function tryGetSubType(module: ReactPyModule, component: string) {
+  let subComponents: string[] = component.split(".");
+  const rootComponent: string = subComponents[0];
+  let subComponentAccessor: string = rootComponent;
+  let type: any = module[rootComponent];
+
+  subComponents = subComponents.slice(1);
+  for (let i = 0; i < subComponents.length; i++) {
+    const subComponent = subComponents[i];
+    subComponentAccessor += "." + subComponent;
+    type = type[subComponent];
+    if (!type) {
+      console.error(
+        `Component ${rootComponent} does not have subcomponent ${subComponentAccessor}`,
+      );
+      break;
+    }
+  }
+  return type;
+}
+
```

Comment:
1. No reason to prefix function name with with `try` here.
   - Rename to something else (maybe `getModuleType`?)
2. Add a comment describing what the function does.
3. Add typescript type hint for return type

Reply:
I've addressed all these. As I mentioned above, I went with `getComponentFromModule` as the new function name and moved some additional logic from the `createImportSourceElement` function to this one. Take a look.